lib/ansible/modules/cloud/amazon/ec2_key.py

```diff
@@ -62,44 +62,39 @@
 '''
 
 EXAMPLES = '''
-# Note: None of these examples set aws_access_key, aws_secret_key, or region.
-# It is assumed that their matching environment variables are set.
+# Note: These examples do not set authentication details, see the AWS Guide for details.
 
-# Creates a new ec2 key pair named `example` if not present, returns generated
-# private key
-- name: example ec2 key
+- name: create a new ec2 key pair, returns generated private key
   ec2_key:
-    name: example
+    name: my_keypair
 
-# Creates a new ec2 key pair named `example` if not present using provided key
-# material.  This could use the 'file' lookup plugin to pull this off disk.
-- name: example2 ec2 key
+- name: create key pair using provided key_material
   ec2_key:
-    name: example2
+    name: my_keypair
     key_material: 'ssh-rsa AAAAxyz...== me@example.com'
-    state: present
 
-# Given example2 is already existing, the key will not be replaced because the
-# force flag was set to `false`
-- name: example2 ec2 key
+- name: create key pair using key_material obtained using 'file' lookup plugin
   ec2_key:
-    name: example2
-    key_material: 'ssh-rsa AAAAxyz...== me@example.com'
-    force: false
-    state: present
+    name: my_keypair
+    key_material: "{{ lookup('file', '/path/to/public_key/id_rsa.pub') }}"
 
-# Creates a new ec2 key pair named `example` if not present using provided key
-# material
-- name: example3 ec2 key
+- name: another way to create a key pair using key_material from a file
```

Comment:
Not convinced we need this example, particularly with `with_` going away.

Reply:
Sorry, I didn't know about `with_` going away. I have replaced the already existing `with_file` example with a `lookup('file', 'id_rsa.pub')` example.